ProcessLib/ThermoMechanics/ThermoMechanicsProcess.cpp

```diff
@@ -154,53 +163,13 @@ void ThermoMechanicsProcess<DisplacementDim>::constructDofTable()
             std::move(all_mesh_subsets_single_component),
             // by location order is needed for output
             NumLib::ComponentOrder::BY_LOCATION));
-
-    // Vector of mesh subsets.
-    std::vector<MeshLib::MeshSubset> all_mesh_subsets;
-
-    // Vector of the number of variable components
-    std::vector<int> vec_var_n_components;
     if (_use_monolithic_scheme)
     {
-        // Collect the mesh subsets in a vector for each variables' components.
-        for (ProcessVariable const& pv : _process_variables[0])
-        {
-            std::generate_n(std::back_inserter(all_mesh_subsets),
-                            pv.getNumberOfComponents(),
-                            [&]() { return *_mesh_subset_all_nodes; });
-        }
-
-        // Create a vector of the number of variable components
-        for (ProcessVariable const& pv : _process_variables[0])
-        {
-            vec_var_n_components.push_back(pv.getNumberOfComponents());
-        }
-    }
-    else  // for staggered scheme
-    {
-        // Collect the mesh subsets in a vector for each variable components for
-        // the mechanical process.
-        std::generate_n(
-            std::back_inserter(all_mesh_subsets),
-            _process_variables[_process_data.mechanics_process_id][0]
-                .get()
-                .getNumberOfComponents(),
-            [&]() { return *_mesh_subset_all_nodes; });
-
-        // Create a vector of the number of variable components.
-        vec_var_n_components.push_back(
-            _process_variables[_process_data.mechanics_process_id][0]
-                .get()
-                .getNumberOfComponents());
-
-        _sparsity_pattern_with_single_component =
-            NumLib::computeSparsityPattern(
-                *_local_to_global_index_map_single_component, _mesh);
+        return;
```

Comment:
I'd invert the condition, so it is more explicit.
Btw, why is the sparsity pattern computed only for staggered scheme?

Reply:
Changed.

The another sparse pattern is a member of Process, and it is created by calling Process::constructDofTableOfSpecifiedProsessStaggerdScheme(specified_prosess_id)